crates/cairo-lint-core/src/fix.rs

```diff
@@ -180,10 +184,107 @@ impl Fixer {
+    /// * `node` - The `SyntaxNode` representing the loop expression.
+    ///
+    /// # Returns
+    ///
+    /// A `String` containing the transformed loop as a `while` loop, preserving
+    /// the original formatting and indentation.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let mut x = 0;
+    /// loop {
+    ///     if x > 5 {
+    ///         break;
+    ///     }
+    ///     x += 1;
+    /// }
+    /// ```
+    ///
+    /// Would be converted to:
+    ///
+    /// ```
+    /// let mut x = 0;
+    /// while x <= 5 {
+    ///     x += 1;
+    /// }
+    /// ```
+    pub fn fix_loop_break(&self, db: &dyn SyntaxGroup, node: SyntaxNode) -> String {
+        let loop_expr = ExprLoop::from_syntax_node(db, node.clone());
+        let mut condition_text = String::new();
+        let mut loop_body = String::new();
+
+        let indent = node.get_text(db).chars().take_while(|c| c.is_whitespace()).collect::<String>();
+
+        if let Some(Statement::Expr(expr_statement)) = loop_expr.body(db).statements(db).elements(db).first() {
+            if let Expr::If(if_expr) = expr_statement.expr(db) {
+                let condition = if_expr.condition(db);
+                condition_text = condition.as_syntax_node().get_text_without_trivia(db).to_string();
+
+                if condition_text != "true" {
+                    condition_text = Self::invert_condition(&condition_text);
+                }
+
+                for statement in loop_expr.body(db).statements(db).elements(db).iter().skip(1) {
+                    loop_body.push_str(&format!(
+                        "{}    {}\n",
+                        indent,
+                        statement.as_syntax_node().get_text_without_trivia(db)
+                    ));
+                }
+            }
+        }
+
+        format!("{}while {} {{\n{}{}}}\n", indent, condition_text, loop_body, indent)
+    }
+
+    fn invert_condition(condition: &str) -> String {
+        if condition.contains("&&") {
+            condition
+                .split("&&")
+                .map(|part| Self::invert_simple_condition(part.trim()))
+                .collect::<Vec<_>>()
+                .join(" || ")
+        } else if condition.contains("||") {
+            condition
+                .split("||")
+                .map(|part| Self::invert_simple_condition(part.trim()))
+                .collect::<Vec<_>>()
+                .join(" && ")
+        } else {
+            Self::invert_simple_condition(condition)
+        }
+    }
+
+    fn invert_simple_condition(condition: &str) -> String {
+        if condition.contains(">=") {
+            condition.replace(">=", "<")
+        } else if condition.contains("<=") {
+            condition.replace("<=", ">")
+        } else if condition.contains('>') {
+            condition.replace('>', "<=")
+        } else if condition.contains('<') {
+            condition.replace('<', ">=")
+        } else {
+            format!("!({})", condition)
```

Comment:
I'd only keep the `else`, otherwise it's going to be a mess the first line will add `<` that will be replaced in the last `else if`